packages/cicero-core/lib/template.js

```diff
@@ -484,27 +484,30 @@ class Template {
                 template = new Template(packageJsonContents, readmeContents);
 
                 logger.debug(method, 'Adding model files to model manager');
-                template.modelManager.addModelFiles(ctoModelFiles, ctoModelFileNames); // Adds all cto files to model manager
-                logger.debug(method, 'Added model files to model manager');
-                logger.debug(method, 'Adding JavaScript files to script manager');
-                jsScriptFiles.forEach(function (obj) {
-                    let jsObject = template.scriptManager.createScript(obj.name, 'js', obj.contents);
-                    template.scriptManager.addScript(jsObject); // Adds all js files to script manager
-                });
-                logger.debug(method, 'Added JavaScript files to script manager');
-
-                // check the template model
-                template.getTemplateModel();
-
-                logger.debug(method, 'Setting grammar');
-                if (grammar) {
-                    template.setGrammar(grammar);
-                } else {
-                    template.buildGrammar(templatizedGrammar);
-                }
+                template.modelManager.addModelFiles(ctoModelFiles, ctoModelFileNames, true); // Adds all cto files to model manager
+                return template.getModelManager().updateExternalModels().then(() => {
+                    template.getModelManager().validateModelFiles();
```

Comment:
updateExternalModels calls validate internally, so you don't need this.

Reply:
I think we need to make the call to updateExternalModels optional, or every time we create a template (e.g. for parsing) we will hit the network to download CTO files.